WordPress/gutenberg
# Block Editor: Add new `TextAlignmentControl` component

Part of #60762

## What?

This PR adds a new `TextAlignControl` in preparation for adding block-level text alignment UI to the global styles.

## Why?

Adding UI to the global styles in one PR would involve too many changes, so this PR focuses only on adding new components.

A similar component, the [AlignmentControl](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-editor/src/components/alignment-control), already exists, but since this component is [intended to be used in a toolbar](https://github.com/WordPress/gutenberg/blob/d3f13542a222e906fd6ac2d4992d2516823bac20/packages/block-editor/src/components/alignment-control/ui.js#L52), I think it is necessary to create a new component specifically for it.

## How?

- I referenced the implementation of [TextTransformControl](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-editor/src/components/text-transform-control) which has a similar implementation.
- Add Storybook
- Add README

## Testing Instructions

- Run `npm run storybook:dev` and access `http://localhost:50240/?path=/story/blockeditor-textaligncontrol--default`.
- Confirm that changes to the control property are reflected correctly in the component.
- Confirm that the `onChange` action is called correctly when the button is clicked.
- Check the [README](https://github.com/WordPress/gutenberg/blob/block-editor/add-text-align-control/packages/block-editor/src/components/text-align-control/README.md).

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/728bb5d8-31d9-4efd-92a3-2b26abe937c0
